File: redsheet-portal/src/app/database-models/v-audit-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vAuditRecord {
    constructor(vAuditRecord?: vAuditRecord )  {
        let properties = [ "auditRecordId", "auditRequestId", "eventType", "recordId", "recordName", "tableName",];
        for (let property of properties)
            if (vAuditRecord && vAuditRecord[property])
                this[property] = vAuditRecord[property];
    }
 
	auditRecordId : number =   0 ;
 
	auditRequestId : number =   undefined;
 
    @required()
    @maxLength(9)
	eventType : string =   undefined;
 
    @range(1,2147483647)
	recordId : number =   undefined;
 
    @required()
	recordName : string =   undefined;
 
    @required()
    @maxLength(50)
	tableName : string =   undefined;


}

File: RedSheet.DbEntities/Models/vAuditRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vAuditRecords", Schema = "dbo")]
	[Serializable]
	public partial class vAuditRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int AuditRecordId { get; set; } 

	public Nullable<int> AuditRequestId { get; set; } 
 
	[Required] 
	[MaxLength(9)]
	public string EventType { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int RecordId { get; set; } 
 
	[Required]
	public string RecordName { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string TableName { get; set; } 

	public vAuditRecord()
	{
    }

    }
}

